Space-Zoologist/Assets/Scripts/UI/Store/SellingManager.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class SellingManager : MonoBehaviour
+{
+    GridSystem gridSystem = default;
+    TileSystem tileSystem = default;
+    [SerializeField] PauseManager PauseManager = default;
+    [SerializeField] MenuManager MenuManager = default;
+    [SerializeField] Inspector Inspector = default;
+    [SerializeField] PlayerBalance PlayerBalance = default;
+    [SerializeField] LevelDataReference LevelDataReference = default;
+    public bool IsSelling { get; private set; }
+
+    public void OnToggleSell()
+    {
+        if (!IsSelling)
+        {
+            this.PauseManager.TryToPause();
+            if (Inspector.IsInInspectorMode)
+            {
+                Inspector.CloseInspector();
+            }
+            if (MenuManager.IsInStore)
+            {
+                MenuManager.CloseStore();
+            }
+        }
+        IsSelling = !IsSelling;
+    }
+
+    public void StopSelling() {
+        IsSelling = false;
+    }
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        IsSelling = false;
+        gridSystem = FindObjectOfType<GridSystem>();
```

Comment:
FindObjectOfType should typically be used if there's no other easy way to get a reference to another object--it's an expensive search and can usually be replaced with better design. It's also fine to use when prototyping and trying to get functionality working, but should be refactored out if possible. In this case, we can just add the GridSystem and TileSystem as serialized fields.

Reply:
Addressed.